test/Land/Model/haverkamp_test.jl

```diff
@@ -0,0 +1,348 @@
+
+    # When using ClimateMachine, we need to define a function that sets the initial
+    # state of our model run.
+
+    function init_soil_water!(land, state, aux, coordinates, time)
+        FT = eltype(state)
+        state.soil.water.ϑ_l = FT(land.soil.water.initialϑ_l(aux))
+        state.soil.water.θ_ice = FT(land.soil.water.initialθ_ice(aux))
+
+        #    state.ρu = SVector{3, FT}(0, 0, 0) might be a useful ref later for how to initialize vectors.
+    end
+
+
+
+
+    ClimateMachine.init(; disable_gpu = true)
+
+    const clima_dir = dirname(dirname(pathof(ClimateMachine)))
+
+    soil_heat_model = PrescribedTemperatureModel{FT}()
+
+    soil_param_functions = SoilParamFunctions(
+        porosity = 0.495,
+        Ksat = 0.0443 / (3600 * 100),
+        S_s = 1e-3,
+    )
+    # Keep in mind that what is passed is aux⁻
+    # Fluxes are multiplied by ẑ (normal to the surface, -normal to the bottom,
+    # where normal point outs of the domain.)
+    surface_state = (aux, t) -> FT(0.494)
+    bottom_flux = (aux, t) -> FT(aux.soil.water.K * 1.0)
+    ϑ_l0 = (aux) -> FT(0.24)
+
+    soil_water_model = SoilWaterModel(
+        FT;
+        moisture_factor = MoistureDependent{FT}(),
+        hydraulics = Haverkamp{FT}(),
+        initialϑ_l = ϑ_l0,
+        dirichlet_bc = Dirichlet(
+            surface_state = surface_state,
+            bottom_state = nothing,
+        ),
+        neumann_bc = Neumann(surface_flux = nothing, bottom_flux = bottom_flux),
+    )
+
+    m_soil = SoilModel(soil_param_functions, soil_water_model, soil_heat_model)
+    sources = ()
+    m = LandModel(
+        param_set,
+        m_soil;
+        source = sources,
+        init_state_conservative = init_soil_water!,
+    )
+
+
+    N_poly = 5
+    nelem_vert = 10
+
+    # Specify the domain boundaries
+    zmax = FT(0)
+    zmin = FT(-1)
+
+    driver_config = ClimateMachine.SingleStackConfiguration(
+        "LandModel",
+        N_poly,
+        nelem_vert,
+        zmax,
+        param_set,
+        m;
+        zmin = zmin,
+        numerical_flux_first_order = CentralNumericalFluxFirstOrder(),
+    )
+
+    t0 = FT(0)
+    timeend = FT(60 * 60 * 24)
+
+    dt = FT(6)
+
+    solver_config = ClimateMachine.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config,
+        ode_dt = dt,
+    )
+    mygrid = solver_config.dg.grid
+    Q = solver_config.Q
+    aux = solver_config.dg.state_auxiliary
+
+    ClimateMachine.invoke!(solver_config)
+    t = ODESolvers.gettime(solver_config.solver)
+    state_vars = SingleStackUtils.get_vars_from_nodal_stack(
+        mygrid,
+        Q,
+        vars_state_conservative(m, FT),
+    )
+    aux_vars = SingleStackUtils.get_vars_from_nodal_stack(
+        mygrid,
+        aux,
+        vars_state_auxiliary(m, FT),
+    )
+    all_vars = OrderedDict(state_vars..., aux_vars...)
+    all_vars["t"] = [t]
+
+    # Compare with Bonan simulation data at 1 day.
+    bonan_moisture = reverse([
```

Comment:
Just to keep track, we should probably track this with Artifacts, as is done in Thermodynamics [here](https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Common/Thermodynamics/data_tests.jl).

Reply:
ok! I will add this to the list, now there are three items:
consider making a wrapper for functions like pressure head and hydraulic conductivity
add Bonan data using Artifacts
make orientation an attribute of the LandModel. Use init_aux and pass in orientation structure to that to generate auxiliary variables.